mart/attack/adversary.py

```diff
@@ -5,45 +5,44 @@
 # agreement between Intel Corporation and you.
 #
 
+import abc
 from collections import OrderedDict
 
 import torch
 
-from .callbacks import Callback
-
 __all__ = ["Adversary", "NoAdversary"]
 
 
-class AdversaryCallbackHookMixin(Callback):
+class AdversaryCallbackHookMixin(abc.ABC):
```

Comment:
A mixin is not an ABC! So this should just inherit from object or be renamed.

Reply:
The real purpose of this mixin is to add methods to whatever class so that class can fire the callbacks.